plugins/aea-ledger-ethereum/aea_ledger_ethereum/ethereum.py

```diff
@@ -1206,6 +1214,180 @@ def is_valid_address(cls, address: Address) -> bool:
         """
         return Web3.isAddress(address)
 
+    def _call(
+        self,
+        contract_interface: Dict[str, str],
+        method_name: str,
+        *method_args: Any,
+    ) -> Optional[JSONLike]:
+        """Call method."""
+        contract = self.get_contract_instance(contract_interface)
+        method = getattr(contract.functions, method_name)
+        result = method(*method_args).call()
+        return result
+
+    def _prepare_tx(  # pylint: disable=too-many-arguments
+        self,
+        contract_interface: Dict[str, str],
+        sender_address: str,
+        method_name: str,
+        *method_args: Any,
+        eth_value: int = 0,
+        gas: Optional[int] = None,
+        gas_price: Optional[int] = None,
+        max_fee_per_gas: Optional[int] = None,
+        max_priority_fee_per_gas: Optional[int] = None,
+    ) -> Optional[JSONLike]:
+        """Prepare tx method."""
+        contract = self.get_contract_instance(contract_interface)
+        method = getattr(contract.functions, method_name)
+        tx = method(*method_args)
+        tx = self._build_transaction(
+            sender_address,
+            tx,
+            eth_value,
+            gas,
+            gas_price,
+            max_fee_per_gas,
+            max_priority_fee_per_gas,
+        )
+        return tx
+
+    def _build_transaction(  # pylint: disable=too-many-arguments
+        self,
+        sender_address: str,
+        tx: Any,
+        eth_value: int = 0,
+        gas: Optional[int] = None,
+        gas_price: Optional[int] = None,
+        max_fee_per_gas: Optional[int] = None,
+        max_priority_fee_per_gas: Optional[int] = None,
+    ) -> Optional[JSONLike]:
+        """Build transaction method."""
+        nonce = self.api.eth.get_transaction_count(sender_address)
+        tx_params = {
+            "nonce": nonce,
+            "value": eth_value,
+        }
+        if gas is not None:
+            tx_params["gas"] = gas
+        if gas_price is not None:
+            tx_params["gasPrice"] = gas_price
+        if max_fee_per_gas is not None:
+            tx_params["maxFeePerGas"] = max_fee_per_gas  # pragma: nocover
+        if max_priority_fee_per_gas is not None:
+            tx_params[
+                "maxPriorityFeePerGas"
+            ] = max_priority_fee_per_gas  # pragma: nocover
+        if (
+            gas_price is None
+            and max_fee_per_gas is None
+            and max_priority_fee_per_gas is None
+        ):
+            gas_data = self.try_get_gas_pricing()
+            if gas_data:
+                tx_params.update()  # pragma: nocover
+        tx = tx.buildTransaction(tx_params)
+        return tx
+
+    def get_tx_transfer_logs(  # pylint: disable=too-many-arguments,too-many-locals
+        self,
+        contract_interface: Dict[str, str],
+        tx_hash: str,
+        target_address: Optional[str] = None,
+    ) -> JSONLike:
```

Comment:
> Eg this (https://github.com/valory-xyz/consensus-algorithms/blob/main/packages/valory/contracts/offchain_aggregator/contract.py#L230-L307) can be implemented once in aea ethereum ledger plugin and then exposed on contract base class for usage in each of the contract packages.

How can I expose these functions on the contract base? It seems every function in the plugin is used directly from `ledger_api`, not from contract.